File: M1files/repl.py
from M1files.lexer import Lexer
from M1files.parser import Parser
from M1files.ast import Program
from M1files.tokens import(
    Token,
    TokenType
)

EOF_TOKEN:Token=Token(TokenType.EOF, '')
def start_repl()->None:
    while(source:=input('>> '))!='close()':
        lexer:Lexer=Lexer(source)
        parser:Parser=Parser(lexer)
        program:Program=parser.parse_program()
        
        print(program)

File: M1files/Repl.java
package M1files;

public class Repl {
    Token EndOfFile_TOKEN = new Token(TokenType.EOF, "");

    public void start_repl(){
        String source;
        while(true){
            System.out.print(">> ");
            source = System.console().readLine();
            if(source.equals("exit()")) break;

            Lexer lexer = new Lexer(source);
            Parser parser = new Parser(lexer);
            Program program = parser.parseProgram();
            Object evaluated = Evaluator.evaluate(program);
            if(evaluated != null) 
                System.out.println(evaluated.inspect());
        }
    }
}
